engine/src/main/java/org/terasology/config/InternetConfig.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config;
+
+import com.google.common.collect.Maps;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import gnu.trove.set.TIntSet;
+import gnu.trove.set.hash.TIntHashSet;
+import org.terasology.naming.Name;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Configuration file for all the internet BS you can think of.
+ */
+// TODO make thread safe and stuff
+public class InternetConfig {
+    /**
+     * All the TCP whitelists. Keyed by module ID.
+     */
+    private Map<Name, Hosts> tcpWhitelists = Maps.newTreeMap();
```

Comment:
AFAIK there's some sort of `TObjectObjectMap` class. Might be worth using alongside `TIntSet`.